### ISSUE TITLE
FLAC recording not working on macOS

### ISSUE THREAD

**Issue by svintuss** [mixxxdj/mixxx#10880]
### Bug Description

When FLAC output file format is selected in `Preferences > Recording` pressing `Options > Record Mix` produces an error window: `Failed to apply the selected settings.` Changing Bit depth or Compression level has no effect on the issue.

Recording in other formats (apart from HE-AAC and HE-AACv2) works fine.

### Version

2.3.3, 2.4-alpha-1299-gbc6ab43dc9

### OS

macOS Monterey

**Comment by ronso0** [comment]
Can you play Flac files?

**Comment by ronso0** [comment]
Maybe the log contains some useful information.
Please start Mixxx with `--developer` argument  and copy/paste here what is being printed:
* open preferences, go to Recording
* select Flac if it isn't already, Apply
* close preferences
* start Recording (Options menu, Rec toggle in GUI or Ctrl+R)

**Comment by svintuss** [comment]
I can play Flac files without issue. 

This is what I get when I try to start Flac recording:
```
debug [Main] Split time is: 4800
debug [Main] Recordings folder set to "/Users/user/Audio/Mixxx/Recordings"
debug [EngineSideChain 1] Error initializing Wave encoding. sf_open_virtual returned: File contains data in an unimplemented format.
debug [EngineSideChain 1] Setting record flag to: OFF
debug [Main] Recordings folder set to "/Users/user/Audio/Mixxx/Recordings"
```

**Comment by daschuer** [comment]
An issue can be your soundcard preferences.
The recording is using the same samplerate.
What did you pick there?

**Comment by svintuss** [comment]
@daschuer I use 44,1 kHz for both input and output on my all of my sound cards. Mixxx's sample rate is also set to 44100 Hz. 
I have this error even if I set all input and output devices in `Sound hardware` preference pane to `---`.

**Comment by zfhrp6** [comment]
I have the same issue on Windows in Mixxx ver.2.4.0.

The same error window and the same log are given.

I check following,

* In 2.3.6, this doesn't happen.
* In 2.4.0, happens.

**Comment by JoergAtGithub** [comment]
https://mixxx.discourse.group/t/recording-failure-with-flac-in-2-4-0/

**Comment by daschuer** [comment]
FLAC encoding is not enabled in our build environment. I will take a look how to fix that.

**Comment by daschuer** [comment]
Here is a PR with a new build environment that should allow FALC and OGG recording

**Comment by JoergAtGithub** [comment]
Fixed in #13154
